File: backend/contracts/validators.py
import re
from django.core.exceptions import ValidationError

def validate_cpf(value):
    cpf = ''.join(re.findall(r'\d', str(value)))
    if not cpf or len(cpf) != 11 or cpf == cpf[0] * 11:
        raise ValidationError('CPF inválido.')
    sum_ = sum(int(cpf[i]) * (10 - i) for i in range(9))
    digit1 = (sum_ * 10) % 11
    if digit1 == 10: digit1 = 0
    if digit1 != int(cpf[9]): raise ValidationError('CPF inválido.')
    sum_ = sum(int(cpf[i]) * (11 - i) for i in range(10))
    digit2 = (sum_ * 10) % 11
    if digit2 == 10: digit2 = 0
    if digit2 != int(cpf[10]): raise ValidationError('CPF inválido.')
    return value

def validate_rg(value):
    rg = ''.join(re.findall(r'[\dXx]', str(value).upper()))
    if not rg or len(rg) < 7 or len(rg) > 9:
         raise ValidationError('Formato de RG inválido.')
    return value

def validate_cnpj(value): # Novo validador ABNT
    cnpj = ''.join(re.findall(r'\d', str(value)))
    if len(cnpj) != 14 or cnpj == cnpj[0] * 14:
        raise ValidationError('CNPJ inválido.')
    # (Lógica de validação de CNPJ mais complexa pode ser adicionada aqui)
    return value

File: frontend/src/utils/validators.ts
export const validaCPF = (cpf: string): boolean => {
  cpf = cpf.replace(/[^\d]+/g, '');
  if (cpf.length !== 11 || /^(\d)\1+$/.test(cpf)) return false;
  let sum = 0;
  for (let i = 0; i < 9; i++) sum += parseInt(cpf.charAt(i)) * (10 - i);
  let digit = (sum * 10) % 11;
  if (digit === 10) digit = 0;
  if (digit !== parseInt(cpf.charAt(9))) return false;
  sum = 0;
  for (let i = 0; i < 10; i++) sum += parseInt(cpf.charAt(i)) * (11 - i);
  digit = (sum * 10) % 11;
  if (digit === 10) digit = 0;
  if (digit !== parseInt(cpf.charAt(10))) return false;
  return true;
};

export const validaRG = (rg: string): boolean => {
  const rgLimpo = rg.replace(/[^\dX]+/gi, '');
  return rgLimpo.length >= 7 && rgLimpo.length <= 9;
};

// Adicione validação de CNPJ se necessário
export const validaCNPJ = (cnpj: string): boolean => {
    const digits = cnpj.replace(/\D/g, '');
    if (digits.length !== 14 || /^(\d)\1+$/.test(digits)) return false;
    // Lógica de validação de CNPJ (mais complexa) pode ser adicionada aqui
    return true;
}